agent/lib/kontena/rpc_server.rb

```diff
@@ -58,7 +59,7 @@ def handle_request(ws_client, message)
     # @param [WebsocketClient] ws_client
     # @param [Array, Hash] msg
     def send_message(ws_client, msg)
-      ws_client.send_message(MessagePack.dump(msg).bytes)
+      ws_client.async.send_message(MessagePack.dump(msg).bytes)
```

Comment:
The `ws_client.send_message` can block or raise errors... no point calling it sync, because the rpc server can't really do anything about that. They will get logged as warnings by the websocket client.